src/v3/src/hooks/useAutoFocus.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { MutableRef, useEffect, useRef } from 'preact/hooks';
+
+type InteractiveElement =
+| HTMLInputElement
+| HTMLButtonElement
+| HTMLTextAreaElement
+| HTMLSelectElement
+| HTMLAnchorElement;
+
+export const useAutoFocus = (
+  focus?: boolean,
+): MutableRef<InteractiveElement | undefined> => {
```

Comment:
Ok, there are some type fixes I think would improve this hook (so you don't have to typecast). I'm going to try taking a generic and propose it here in a bit.

Reply:
awesome sounds good